workflow/controller/operator.go

```diff
@@ -2386,6 +2388,15 @@ func (woc *wfOperationCtx) estimateNodeDuration(nodeName string) wfv1.EstimatedD
 	return woc.getEstimator().EstimateNodeDuration(nodeName)
 }
 
+func (woc *wfOperationCtx) hasNodeWithAgentPod() bool {
+	for _, node := range woc.wf.Status.Nodes {
+		if node.Type == wfv1.NodeTypePlugin || node.Type == wfv1.NodeTypeHTTP {
```

Comment:
I don't believe it's just `plugin` and `http` either, per https://github.com/argoproj/argo-workflows/issues/12659#issuecomment-1939189088.

`controller/agent.go` actually does not seem to define the the template types exclusively, so maybe we should instead check if the Pod contains the agent -- the `isAgentPod` function already exists there.

Reply:
Each workflow involving http/plugin template has an agent pod, see [#12659 comment].(https://github.com/argoproj/argo-workflows/issues/12659#issuecomment-1956125777)
The `isAgentPod` function determine whether a pod is the agent type based on its name, so it can not be used here.

Reply:
This function can be used to determine whether the creation of `AgentPod` is necessary.
https://github.com/argoproj/argo-workflows/blob/7ba20fea08e40a5792a1d25011ee8c12889960c5/workflow/controller/taskset.go#L59-L61

Reply:
The above done in #13294